src/treemodel/rootelement.js

```diff
@@ -19,16 +19,35 @@ export default class RootElement extends Element {
 	 *
 	 * @param {engine.treeModel.Document} doc {@link engine.treeModel.Document} that is an owner of the root.
 	 * @param {String} name Node name.
+	 * @param {String} rootName Root name inside parent {@link engine.treeModel.Document}.
 	 */
-	constructor( doc, name ) {
+	constructor( doc, name, rootName ) {
```

Comment:
I do not find it necessarily. If someone creates root in the document using `createRoot`, this method will create `RootElement` with the same name. There is no other way to create root in the `Document`, so there is not way to desynchronise these names. Also If someone want to create a root element, for instance controller, temporarily, for some reason, then, why should we prevent it?